src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+  publishing new versions.
+{{site.alert.end}}
+
+
+## Configure automated publishing from GitHub Actions on pub.dev
+
+### Check for required `pub.dev` permissions
+
+To automate publication from GitHub Actions to `pub.dev`, you need one
+of the following roles:
+
+* An _uploader_ on the package
+* An _admin_ of the publisher if a publisher owns the package
+
+### Configure automated publishing
+
+To enable `pub.dev` to accept GitHub Action publishing, complete the
+following steps.
+
+1. Navigate to the **Admin** tab (`pub.dev/packages/<package>/admin`).
+1. Find the **Automated publishing** section.
+
+   ![pub.dev Publishing from GitHub Actions form](pub-dev-gh-setup.png)
+
+1. Click **Enable publishing from GitHub Actions**.
+   Additional fields display when you select this checkbox.
+1. Write the GitHub repository in the **Repository** field.
+
+   A _repository_ contains all of your package's files on GitHub.
+   This field expects a format of `<organization>/<repository>`.
+   For example: If the URL for your repository resembles
+   `https://github.com/dart-lang/pana`, write `dart-lang/pana` in the
+   **Repository** field.
+
+1. Write the _`tag-pattern`_ into the **Tag pattern** field.
+
+   A `tag-pattern` represents the version of the package.
+   This field expects a string that includes the literal
+   `{% raw %}{{version}}{% endraw %}`.
+   Github Actions can publish your package only if a push of a git tag
+   matching this `tag-pattern` triggers the Action.
+   The package's `pubspec.yaml` file requires the `version` key.
+   This key's value must match the `{% raw %}{{version}}{% endraw %}`
+   in the `tag-pattern`.
+
+1. Click **Update**.
+
+<!-- **Example:** You wrote your `tag-pattern` as `v{{version}}`.
+You then tag this commit and push it.
+
+```shell
+git tag v1.2.3 && git push v1.2.3
+```
+
+This push triggers Github Actions to publish version `1.2.3` of
+your package. -->
+
+If your repository contains multiple packages, give each a separate
+`tag-pattern`. Consider using a `tag-pattern` like
+`my_package_name-v``{% raw %}{{version}}{% endraw %}` for a package
+named `my_package_name`.
+
+## Prepare your package repository for automated publishing
+
+
+After you enable automated publishing from GitHub Actions on pub.dev,
+you need to make two changes to your package repository.
+
+1. Modify the package's `pubspec.yaml` file.
+2. Create a GitHub Action workflow file.
+
+### Update pubspec.yaml with the package version
+
+The `pubspec.yaml` file must reflect the current version of the package.
+
+1. Navigate to the root of your local copy of your package repository.
+1. Open the `pubspec.yaml` file using your preferred editor.
+1. Update or add the `version` key with the correct version number.
+   The version number must follow [semantic versioning][27].
+
+    ```yaml
+    package: my_package_name
+    version: 1.2.3  # must match the version number used in the git tag
+    environment:
+      sdk: ^2.18.0
+    ...
```

Comment:
This has to do with publishing a specific version. Not configuration.
I suggest we drop it